### ISSUE TITLE
`docker builder prune` no longer respects `--filters`

### ISSUE THREAD

**Issue by alexander-hirth** [moby/moby#43734]
Docker 22.06.0-beta.0 broke `docker builder prune --filter "until=XYZ". I get the following on my fedora 36 system:

```
[alex@ad ~]$ docker builder prune -f --filter "until=24h" --verbose
Total:  0B
[alex@ad ~]$ docker builder prune
WARNING! This will remove all dangling build cache. Are you sure you want to continue? [y/N] y
ID                                              RECLAIMABLE     SIZE            LAST ACCESSED
u5752nigmrh4tlyboso5waiei                       true            98.09MB         46 hours ago
rinmtsj7ckgfc1kht2z1zyluw                       true    60.83MB         45 hours ago
indan1ushjvqe4kfkrugpy68d                       true    653.8MB         44 hours ago
94qdrqmlhi77068lih5cvcpl4                       true    0B              32 hours ago
su9a6wrhefbn0xpfawhb4k6p4                       true    653.8MB         44 hours ago
odkwfoatyk73jqztnc6jocrod                       true    98.1MB          33 hours ago
yypopqkfk8qsulg2ufu100d9d                       true    0B              44 hours ago
ndtg45xjxxrhu5nog55bz1uw4                       true    101.4MB         45 hours ago
p67kbju3eefseaqexvmy1tsm9                       true    98.1MB          44 hours ago
nble8o6rougry4myx41cyzey5                       true    0B              45 hours ago
u05q4vx2889aevmuruzn5ty3v                       true    101.4MB         44 hours ago
q56k51l8vvwkvchrn5g2lk5lo                       true    101.4MB         33 hours ago
9gouwora0chhzyqkws6f4y9q8                       true    78.36MB         46 hours ago
...
```

The command `docker builder prune -f --filter "until=24h"` used to work in a version of Docker that was shipped with Fedora 35.

**Output of `docker version`:**

```
Client: Docker Engine - Community
 Version:           22.06.0-beta.0
 API version:       1.42
 Go version:        go1.18.3
 Git commit:        3e9117b
 Built:             Fri Jun  3 17:59:03 2022
 OS/Arch:           linux/amd64
 Context:           default

Server: Docker Engine - Community
 Engine:
  Version:          22.06.0-beta.0
  API version:      1.42 (minimum version 1.12)
  Go version:       go1.18.3
  Git commit:       38633e7
  Built:            Fri Jun  3 17:56:21 2022
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.6.6
  GitCommit:        10c12954828e7c7c9b6e0ea9b0c02b01407d3ae1
 runc:
  Version:          1.1.2
  GitCommit:        v1.1.2-0-ga916309
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

**Output of `docker info`:**

```
Client:
 Context:    default
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc.)
    Version:  v0.8.2
    Path:     /usr/libexec/docker/cli-plugins/docker-buildx
  compose: Docker Compose (Docker Inc.)
    Version:  v2.6.0
    Path:     /usr/libexec/docker/cli-plugins/docker-compose
  scan: Docker Scan (Docker Inc.)
    Version:  v0.17.0
    Path:     /usr/libexec/docker/cli-plugins/docker-scan

Server:
 Containers: 1
  Running: 1
  Paused: 0
  Stopped: 0
 Images: 29
 Server Version: 22.06.0-beta.0
 Storage Driver: overlay2
  Backing Filesystem: xfs
  Supports d_type: true
  Using metacopy: false
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 1
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 io.containerd.runtime.v1.linux runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 10c12954828e7c7c9b6e0ea9b0c02b01407d3ae1
 runc version: v1.1.2-0-ga916309
 init version: de40ad0
 Security Options:
  seccomp
   Profile: builtin
 Kernel Version: 5.18.5-200.fc36.x86_64
 Operating System: Fedora Linux 36 (Thirty Six)
 OSType: linux
 Architecture: x86_64
 CPUs: 16
 Total Memory: 23.47GiB
 Name: ad.vault13
 ID: BK57:HJG6:X5WD:PFIC:PPAZ:GWGB:FPRB:KHVW:6W6G:AXG6:FZ3L:ZLVX
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

I installed docker from Fedora's repositories at baseurl=https://download.docker.com/linux/fedora/$releasever/$basearch/stable.

**Comment by thaJeztah** [comment]
@tonistiigi @crazy-max PTAL

**Comment by crazy-max** [comment]
Tracked upstream: https://github.com/docker/buildx/issues/400

**Comment by alexander-hirth** [comment]
@crazy-max, is docker/buildx#400 really the same bug? It has reports that `--filter` did not work in Nov., 2021. Yet, it did work for me when I was on Fedora 35 and used Docker that shipped with it.

**Comment by thaJeztah** [comment]
This should be fixed in docker buildx v0.9.0 and up (through https://github.com/docker/buildx/pull/1252)
